os/board/Kconfig

```diff
@@ -300,6 +300,15 @@ config BOARD_DUMP_UART
 		If selected it will allow sending RAM and FS dump via UART.
 endif
 
+config SYSTEM_REBOOT_REASON
```

Comment:
Code and header are in arch, but config is in board. It looks strange. Let's move this into arch.